h2o-py/h2o/sklearn/wrapper.py

```diff
@@ -0,0 +1,487 @@
+                         target_param='y',
+                         result_conversion='auto',
+                         converter=_to_h2o_frame,
+                         reverter=_to_numpy):
+    """
+    :param frame_params:
+    :param target_param:
+    :param result_conversion:
+    :param converter:
+    :param reverter:
+    :return:
+    """
+    assert result_conversion in ('auto', True, False)
+
+    def decorator(fn):
+        """
+        A decorator that can be applied to estimator methods, to support the consumption of non-H2OFrame datasets
+        :param fn: the function to be decorated
+        :return: a new function that will convert X, and y parameters before passing them to the original function.
+        """
+        sig = signature(fn)
+        has_self = 'self' in sig.parameters
+        assert any(arg in sig.parameters for arg in frame_params), \
+            "@expect_h2o_frames decorator should be applied to methods taking at least one of {} parameters.".format(frame_params)
+
+        @wraps(fn)
+        def wrapper(*args, **kwargs):
+            """
+            the logic executed before fn: converts params to H2OFrame if needed
+            and possibly backwards before returning result
+            :param args:
+            :param kwargs:
+            :return:
+            """
+            _args = sig.bind(*args, **kwargs).arguments
+            classifier = False
+            estimator_conversion = None
+            self = {}
+            frame_info = None
+            if has_self:
+                self = _args.get('self', None)
+                if isinstance(self, BaseEstimatorMixin):
+                    classifier = self.is_classifier()
+                    estimator_conversion = self._data_conversion_mode()
+                    frame_info = getattr(self, '_frame_params')
+
+            if hasattr(self, '_before_method') and callable(self._before_method):
+                self._before_method(fn=fn, input=_args)
+
+            converted = False
+            for fr in frame_params:
+                if fr in sig.parameters:
+                    as_factor = classifier and fr == target_param
+                    frame_info = frame_info if fr != target_param else None
+                    converted = _convert(converter, fr, _args, as_factor=as_factor, frame_params=frame_info) or converted
+
+            result = fn(**_args)
+            result = _revert(reverter, result, converted=converted,
+                             estimator_conversion=estimator_conversion,
+                             decorator_conversion=result_conversion)
+
+            if hasattr(self, '_after_method') and callable(self._after_method):
+                self._after_method(fn=fn, output=result)
+
+            return result
+
+        return wrapper
+
+    return decorator
+
+
+class BaseEstimatorMixin(object):
+
+    _classifier_distributions = ('bernoulli', 'binomial', 'quasibinomial', 'multinomial')
```

Comment:
In h2o core, you can set distribution to `bernoulli`, `quasibinomial` and `multinomial` to get classifier, no `binomial` option.

Reply:
actually, GLM  is the exception: it is using `binomial` instead of `bernoulli` for its `family` parameter :(

Reply:
yeah, GLM has it's own enum (which is sometimes a good thing but not in this case ;))